### PR TITLE
#325: redesign of Android test 

### PR DESCRIPTION
**Pull request by TikhomirovSergey**
## Change list

- fix of TouchAction. Instances of the TouchAction are reusable now

- redesign of Android test. They are speedy and more stable now 
 
## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Details

1. It was imposible to re-use instances of the TouchAction. Proposed changes should improve it.

2. Enhanced test coverage. More tests and less time.

### CODE REVIEW

**Comment by TikhomirovSergey** [comment]
@SrinivasanTarget Can you review this PR.

@Jonahss @bootstraponline you can take a look at this PR too if you want :)

File: src/test/java/io/appium/java_client/android/AndroidSearchingTest.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+    private static AppiumDriverLocalService service;
+    private static AndroidDriver<AndroidElement> driver;
+
+    @BeforeClass public static void beforeClass() throws Exception {
+        service = AppiumDriverLocalService.buildDefaultService();
+        service.start();
+
+        if (service == null || !service.isRunning())
+            throw new RuntimeException("An appium server node is not started!");
```
Comment (bootstraponline) [inline src/test/java/io/appium/java_client/android/AndroidSearchingTest.java]: This same beforeClass code is duplicated everywhere. maybe refactor this into a method that can be used across multiple classes?

**Comment by TikhomirovSergey** [comment]
@bootstraponline Thanks for the remork. It was improved

File: src/test/java/io/appium/java_client/android/AndroidActivityTest.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        Assert.assertEquals(driver.currentActivity(), ".ContactsListActivity");
+        driver.pressKeyCode(AndroidKeyCode.BACK);
+        Assert.assertEquals(driver.currentActivity(), ".ContactsListActivity");
+    }
+
+    @Test public void startActivityInNewAppTestCaseWithoutClosingApp() {
+        driver.startActivity("io.appium.android.apis", ".accessibility.AccessibilityNodeProviderActivity");
+        Assert.assertEquals(driver.currentActivity(), ".accessibility.AccessibilityNodeProviderActivity");
+        driver.startActivity("com.android.contacts", ".ContactsListActivity", "com.android.contacts", ".ContactsListActivity", false);
+        Assert.assertEquals(driver.currentActivity(), ".ContactsListActivity");
```
Comment (bootstraponline) [inline src/test/java/io/appium/java_client/android/AndroidActivityTest.java]: assertEquals should be statically imported so `Assert.` can be omitted.

**Comment by bootstraponline** [comment]
Code looks good to me. :+1:

**Comment by TikhomirovSergey** [comment]
@bootstraponline The second code style issue was fixed.

**Comment by SrinivasanTarget** [comment]
LGTM

**Comment by bootstraponline** [comment]
:+1:

**Pull request merged**